src/core/worker.js

```diff
@@ -82,6 +84,342 @@ var WorkerTask = (function WorkerTaskClosure() {
+
+  /**
+   * Gets a reader for the range of the PDF data.
+   * @param {number} begin - the start offset of the data.
+   * @param {number} end - the end offset of the data.
+   * @returns {IPDFStreamRangeReader}
+   */
+  getRangeReader: function (begin, end) { return null; },
+
+  /**
+   * Cancels all opened reader and closes all their opened requests.
+   * @param {Object} reason - the reason for cancelling
+   */
+  cancelAllRequests: function (reason) {},
+};
+
+/**
+ * Interface for a PDF binary data reader.
+ *
+ * @interface
+ */
+function IPDFStreamReader() {}
+IPDFStreamReader.prototype = {
+  /** Gets promise when the headers and other meta information is known about
+   * the PDF binary data.
+   * @returns {Promise}
+   */
+  get headersReady() { return null; },
+
+  /**
+   * Gets PDF binary data length. It is defined after the headersReady promise
+   * is resolved.
+   * @returns {number} The data length (or 0 if unknown).
+   */
+  get contentLength() { return 0; },
+
+  /**
+   * Gets ability of the stream to handle range requests. It is defined after
+   * the headersReady promise is resolved.
+   * @returns {boolean}
+   */
+  get disableRange() { return false; },
+
+  /**
+   * Gets ability of the stream to progressively load binary data. It is defined
+   * after the headersReady promise is resolved.
+   * @returns {boolean}
+   */
+  get disableStream() { return false; },
+
+  /**
+   * Requests a chunk of the binary data. The method returns the promise, which
+   * is resolved into object with properties "value" and "done". If the done
+   * is set to true, the stream reached the end, otherwise the value contains
+   * binary data.
+   * @returns {Promise}
+   */
+  read: function () {},
+
+  /**
+   * Cancels all pending read requests and closes the stream.
+   * @param {Object} reason
+   */
+  cancel: function (reason) {},
+
+  /**
+   * Sets or gets the progress callback. The callback can be useful when the
+   * disableStream property of the object is defined as true.
+   */
+  onProgress: null,
+};
+
+/**
+ * Interface for a PDF binary data fragment reader.
+ *
+ * @interface
+ */
+function IPDFStreamRangeReader() {}
+IPDFStreamRangeReader.prototype = {
+  /**
+   * Gets ability of the stream to progressively load binary data.
+   * @returns {boolean}
+   */
+  get disableStream() { return false; },
+
+  /**
+   * Requests a chunk of the binary data. The method returns the promise, which
+   * is resolved into object with properties "value" and "done". If the done
+   * is set to true, the stream reached the end, otherwise the value contains
+   * binary data.
+   * @returns {Promise}
+   */
+  read: function () {},
+
+  /**
+   * Cancels all pending read requests and closes the stream.
+   * @param {Object} reason
+   */
+  cancel: function (reason) {},
+
+  /**
+   * Sets or gets the progress callback. The callback can be useful when the
+   * disableStream property of the object is defined as true.
```

Comment:
Not sure what to put here yet. It's just an alternative solution for the read() method not able to load data in chunks.